exchanges/poloniex/poloniex.go

```diff
@@ -16,915 +17,1002 @@ import (
-	poloniexDepositAddresses     = "returnDepositAddresses"
-	poloniexGenerateNewAddress   = "generateNewAddress"
-	poloniexDepositsWithdrawals  = "returnDepositsWithdrawals"
-	poloniexOrders               = "returnOpenOrders"
-	poloniexTradeHistory         = "returnTradeHistory"
-	poloniexOrderTrades          = "returnOrderTrades"
-	poloniexOrderStatus          = "returnOrderStatus"
-	poloniexOrderCancel          = "cancelOrder"
-	poloniexOrderMove            = "moveOrder"
-	poloniexWithdraw             = "withdraw"
-	poloniexFeeInfo              = "returnFeeInfo"
-	poloniexAvailableBalances    = "returnAvailableAccountBalances"
-	poloniexTradableBalances     = "returnTradableBalances"
-	poloniexTransferBalance      = "transferBalance"
-	poloniexMarginAccountSummary = "returnMarginAccountSummary"
-	poloniexMarginBuy            = "marginBuy"
-	poloniexMarginSell           = "marginSell"
-	poloniexMarginPosition       = "getMarginPosition"
-	poloniexMarginPositionClose  = "closeMarginPosition"
-	poloniexCreateLoanOffer      = "createLoanOffer"
-	poloniexCancelLoanOffer      = "cancelLoanOffer"
-	poloniexOpenLoanOffers       = "returnOpenLoanOffers"
-	poloniexActiveLoans          = "returnActiveLoans"
-	poloniexLendingHistory       = "returnLendingHistory"
-	poloniexAutoRenew            = "toggleAutoRenew"
-	poloniexCancelByIDs          = "/orders/cancelByIds"
-	poloniexTimestamp            = "/timestamp"
-	poloniexWalletActivity       = "/wallets/activity"
-	poloniexMaxOrderbookDepth    = 100
+	poloniexAPIURL   = "https://api.poloniex.com"
+	tradeSpotPath    = "/trade/"
+	tradeFuturesPath = "/futures" + tradeSpotPath
+	marketsPath      = "/markets/"
 )
 
-// Exchange implements exchange.IBotExchange and contains additional specific api methods for interacting with Poloniex
+var (
+	errAddressRequired         = errors.New("address is required")
+	errInvalidWithdrawalChain  = errors.New("invalid withdrawal chain")
+	errInvalidTimeout          = errors.New("invalid timeout")
+	errChainsNotFound          = errors.New("chains not found")
+	errAccountIDRequired       = errors.New("missing account ID")
+	errAccountTypeRequired     = errors.New("account type required")
+	errMarginAdjustTypeMissing = errors.New("margin adjust type invalid")
+	errPositionModeInvalid     = errors.New("invalid position mode")
+)
+
+// Exchange is the overarching type across the poloniex package
 type Exchange struct {
 	exchange.Base
-	details CurrencyDetails
 }
 
-// GetTicker returns current ticker information
-func (e *Exchange) GetTicker(ctx context.Context) (map[string]Ticker, error) {
-	type response struct {
-		Data map[string]Ticker
+// GetSymbol returns symbol detail
+func (e *Exchange) GetSymbol(ctx context.Context, symbol currency.Pair) ([]*SymbolDetail, error) {
+	if symbol.IsEmpty() {
+		return nil, currency.ErrSymbolStringEmpty
 	}
+	var resp []*SymbolDetail
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, unauthEPL, "/markets/"+symbol.String(), &resp)
+}
```

Comment:
🚧 Change request:
* You've changed `GetSymbols` but not `GetSymbol`.
* But I don't actually know how to name this thing, because execution limits are plural
* And I'm not sure if we need this, because we implicitly load all execution limits anyway as [part of Bootstrap](https://github.com/thrasher-corp/gocryptotrader/blob/master/exchanges/exchange.go#L1792), right?
I know we try to support all endpoints, but we'll load all the execution limits anyway, and if the user had cause to get them again, they do have the `GetExecutionLimits` api available.
```suggestion
```

Reply:
With all due respect, I’d kindly ask to leave this section as it is for now.
If anyone isn’t satisfied with it, I’d suggest opening a separate PR to address that.